file_test.go

```diff
@@ -422,15 +430,15 @@ func TestFile__readFromJson(t *testing.T) {
 	}
 	batch := file.Batches[0]
 	batchControl := batch.GetControl()
-	if batchControl.EntryAddendaCount != 2 {
+	if batchControl.EntryAddendaCount != 1 {
```

Comment:
@bkmoovio Shouldn't this be `0`? The `ppd-valid.json` file has no addenda records.

Reply:
I'm not sure why, but `Batch.build` calls `entryCount++` (which is now `1 + entry.addendaCount()`). That doesn't seem right?

Reply:
Hi Adam, 

EntryAddendaCount in the Company/Batch Control Record and File Control Record can be confusing as it is not the AddendaCount for an Entry.   It is  (for reference page OR108) the tally for each Entry Detail Record processed and each Addenda Record processed, within either the batch or file, as appropriate.   For a BatchPPD with one EntryDetail , and no Addenda05, EntryAddendaCount  should be 1,  If the one EntryDetail has an Addenda05, EntryAddendaCount would be 2.  Hopefully that helps . (Clear as Mud)

For the second question:

Prior to removing addendumer change, the code set entryCount as follows:

for i, entry := range batch.Entries {
		entryCount = entryCount + 1 + len(entry.Addendum)....

going forward I had added each entry + each Addenda property if the property was defined. Addenda02, len(Addenda05), Addenda98, Addenda99.  it looks like you created a function for that which is great.  I should of.

Reply:
Regarding ++,

https://github.com/golang/go/issues/21263

To funny... :-)

Reply:
Ah, okay. Those names are confusing but it makes sense.

Reply:
well that's a good idea.